src/WebJobs.Extensions.DocumentDB/Bindings/DocumentDBEnumerableBuilder.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Documents;
+using Microsoft.Azure.Documents.Client;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DocumentDB
+{
+    internal class DocumentDBEnumerableBuilder<T> : IAsyncConverter<DocumentDBAttribute, IEnumerable<T>> where T : class
+    {
+        private DocumentDBConfiguration _config;
+
+        public DocumentDBEnumerableBuilder(DocumentDBConfiguration config)
+        {
+            _config = config;
+        }
+
+        public async Task<IEnumerable<T>> ConvertAsync(DocumentDBAttribute attribute, CancellationToken cancellationToken)
+        {
+            DocumentDBContext context = _config.CreateContext(attribute);
+
+            Uri collectionUri = UriFactory.CreateDocumentCollectionUri(context.ResolvedAttribute.DatabaseName, context.ResolvedAttribute.CollectionName);
+
+            List<Document> finalResults = new List<Document>();
+
+            string continuation = null;
+
+            SqlQuerySpec sqlSpec = new SqlQuerySpec
+            {
+                QueryText = context.ResolvedAttribute.SqlQuery,
+                Parameters = context.ResolvedAttribute.SqlQueryParameters ?? new SqlParameterCollection()
+            };
+
+            do
+            {
+                DocumentQueryResponse response = await DocumentDBUtility.RetryAsync(
+                    () => context.Service.ExecuteNextAsync(collectionUri, sqlSpec, continuation),
+                    context.MaxThrottleRetries, codesToIgnore: HttpStatusCode.NotFound);
+
+                if (response == null)
+                {
+                    return null;
+                }
+
+                finalResults.AddRange(response.Results);
+                continuation = response.ResponseContinuation;
+            }
+            while (!string.IsNullOrEmpty(continuation));
+
+            return finalResults.Select(doc => (T)(dynamic)doc);
```

Comment:
`doc` is a `Document` (a DocumentDB Type), which implements `IDynamicMetaObjectProvider`, allowing it to be cast as a `dynamic`. Once you have a `dynamic`, you cast it T. There's a chance it fails (if your properties don't match).